### ISSUE TITLE
`[ReadOnly]Span<T>` fields and properties in a `ref struct` can have a `stackalloc` as a default value

### ISSUE THREAD

**Issue by DaZombieKiller** [dotnet/roslyn#60568]
The following code compiles successfully:
```cs
using System;

ref struct Example
{
    public Span<byte> Field = stackalloc byte[512];
    public Span<byte> Property { get; } = stackalloc byte[512];
    public Example() {}
}
```
([SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLTJguBjGrcD2CyANDCAJYA2APgAIBMAjALABQ7SYABBlrtwCiADygBbAA5UI7AN7tui7nQDM3AMoSoAOwA8wAJ4wIAPm4AxChCoATbgF5eMKNgDWUKlQLZuh4wG0AViYGAF0AbgUlVQ0tPT9TbgAFBAIJCAQYA25ZbgBzCBhw7gBfBycXd09vXyMIIJCIqMUYkXEpCAAKAEockvZ+tiA=))

Neither the field nor the property are valid to access, as they are referencing stack memory from the constructor. Since this construct is wholly unsafe, it should probably be disallowed or at least warned.

**Comment by Mindstyler** [comment]
Yes.
Good to see that the bug report is already filed.

**Comment by RikkiGibson** [comment]
I expected an error here for the same reason we get an error [here](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAcoKgWzczAOwBoYQpkYAfAAQCYBGAWACgaBmAAns4GFOA3q04juXGgBZOAWQAUAezgxOAZSIAeAJaEYAPk4EohAJSDhoi4cKcAvAZhQAxgGsoAGzfzHnbTADaDABsALoA3OYiAL6skUA===).

**Comment by cston** [comment]
@DaZombieKiller, thanks for reporting this issue.

Fixed in https://github.com/dotnet/roslyn/pull/60577.
